src/reducers/SignIn.js

```diff
@@ -0,0 +1,52 @@
+import {
+  SIGNING_USER_IN,
+  USER_SIGN_IN_FULFILLED,
+  USER_SIGN_IN_FAILED,
+  CLEAR_ERROR,
+  SIGNING_IN_REQUEST
+} from '../actions/action.types';
+
+const initialState = {
+  token: '',
+  user: {},
+  signingIn: false,
+  error: '',
+};
+
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    case SIGNING_IN_REQUEST:
+      return {
+        ...state,
+        signingIn: true,
+      };
+    case SIGNING_USER_IN:
```

Comment:
SIGNING_USER_IN is not used. I think it does the same thing with SIGNING_IN_REQUEST